Framework/Algorithms/src/Divide.cpp

```diff
@@ -76,9 +76,15 @@ void Divide::performBinaryOperation(const HistogramData::Histogram &lhs, const d
 
 void Divide::setOutputUnits(const API::MatrixWorkspace_const_sptr lhs, const API::MatrixWorkspace_const_sptr rhs,
                             API::MatrixWorkspace_sptr out) {
+
+  if (lhs->isRaggedWorkspace() && rhs->isRaggedWorkspace()) {
```

Comment:
I think this belongs as part of the if-tree on L84 rather than a separate case. The way that it is here can be overridden by the if-tree starting on L84. I think the logic should be

if both ragged and y-units match -> set distribution and clear y-units
else if y-units match and rhs-blocksize is bigger than one -> set distribution and clear y-units
else -> fall-through case on units